numalogic/registry/redis_registry.py

```diff
@@ -81,24 +84,71 @@ def get_version(key: str) -> str:
         """
         return key.split("::")[-1]
 
-    def __get_model_key(self, latest: bool, version: str, key: str) -> str:
-        if latest:
-            production_key = self.__construct_production_key(key)
-            if not self.client.exists(production_key):
-                raise ModelKeyNotFound(
-                    f"Production key: {production_key}, Not Found !!!\n Exiting....."
-                )
-            model_key = self.client.get(production_key)
-            if not self.client.exists(model_key):
-                raise ModelKeyNotFound(
-                    "Production key = {} is pointing to the key: {} that "
-                    "is missing the redis registry".format(production_key, model_key)
-                )
-        else:
-            model_key = self.__construct_version_key(key, version)
-            if not self.client.exists(model_key):
-                raise ModelKeyNotFound("Could not find model key with key: %s" % model_key)
-        return model_key
+    def _load_from_cache(self, key: str) -> Optional[ArtifactData]:
+        if not self.cache_registry:
+            return None
+        return self.cache_registry.load(key)
+
+    def _save_in_cache(self, key: str, artifact_data: ArtifactData) -> None:
+        if self.cache_registry:
+            self.cache_registry.save(key, artifact_data)
+
+    def _clear_cache(self, key: Optional[str] = None) -> Optional[ArtifactData]:
```

Comment:
is this to clear entire cache or delete a specific key?

Reply:
If key is None i.e we are not passing anything, we clear the whole cache otherwise we just remove the key.